### PR TITLE
add java pom for build java dependency jar

### PR DESCRIPTION
**Pull request by ilovesoup**
Add pom files for java proto dependencies. They are published as jar and referred in other projects.
Note that gogo proto is introduced even though it's not used in Java anyway.
Need to make a gogo proto clean version of it.

---
This change is [Reviewable](https://reviewable.io/reviews/pingcap/kvproto/182)

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+                  <id>copy-resources</id>
+                  <phase>validate</phase>
+                  <goals>
+                     <goal>copy-resources</goal>
+                  </goals>
+                  <configuration>
+                     <outputDirectory>${basedir}/proto/gogoproto</outputDirectory>
+                     <resources>
+                        <resource>
+                           <directory>${GOPATH}/src/github.com/gogo/protobuf/gogoproto</directory>
```
Comment (BusyJay) [inline pom.xml]: GOPATH?

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,152 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<project xmlns="http://maven.apache.org/POM/4.0.0" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance" xsi:schemaLocation="http://maven.apache.org/POM/4.0.0 http://maven.apache.org/xsd/maven-4.0.0.xsd">
+   <modelVersion>4.0.0</modelVersion>
+   <groupId>com.pingcap.tikv</groupId>
+   <artifactId>kvproto</artifactId>
+   <version>0.1.0</version>
```
Comment (BusyJay) [inline pom.xml]: Currently, rust use version `0.0.1`.

File: pom.xml (hunk 1 above)
Comment (ilovesoup) [inline pom.xml]: this is equal to generate-rust.sh
gogoproto introduce special tags and options into proto files and this has to be compiled with   gogoproto dependencies.

File: README.md [hunk 3]
```diff
@@ -6,5 +6,6 @@ Protocol buffer files for TiKV
 + Write your own protocol file in proto folder.
 + Run `make` to generate go and rust code. 
     We generate all go codes in pkg folder and rust in src folder.
++ Run `mvn clean install` to generate and install Java dependencies locally
```
Comment (ilovesoup) [inline README.md]: this is included in tikv-client as submodules. maven install is just for convenience of users who wants to use it this way.

File: pom.xml (hunk 1 above)
Comment (BusyJay) [inline pom.xml]: You can use `${basedir}/_vendor/vendor/github.com/gogo/protobuf/gogoproto`, so java developer doesn't have to install go and setup GOPATH to vendor.

File: pom.xml (hunk 2 above)
Comment (ilovesoup) [inline pom.xml]: changed.

File: pom.xml (hunk 1 above)
Comment (ilovesoup) [inline pom.xml]: it seems tipb (tidb's proto) does not have gogoproto in their source code. I will withdraw this pr and waiting for two repo merge as discussed.

**Comment by ilovesoup** [comment]
it seems tipb (tidb's proto) does not have gogoproto in their source code and can't dealt with the same way. I will withdraw this pr and waiting for two repo merge as discussed.

**Pull request closed without merging**